ENH: Describe execution of MRIQC and how to add the derivatives

celprov opened the pull request:
enh: add justification why QC'ing the data within two weeks of their acquisition
fix: change name of the dataset to what has been implemented (hcph -> hcph-dataset)
enh: instructions to add derivatives subdataset, run MRIQC and push to remote storage
fix: move install datalad instructions before datalad install dataset

@oesteban I'm not too sure about the way I precise the working directory to the container.
Also `mkdocs serve` does not work anymore in my computer (dependency issues of the mkdocs package), so I could not verify if the markdown renders the correct layout. If someone can check for me in the meantime that I fix my mkdocs installation --'

Mentioned in #241

celprov committed:
enh: add justification why QC'ing the data within two weeks of their acquisition
fix: change name of the dataset to what has been implemented (hcph -> hcph-dataset)
enh: instructions to add derivatives subdataset, run MRIQC and push to remote storage
fix: move install datalad instructions before datalad install dataset

oesteban commented on docs/data-management/mriqc.md:
````diff
@@ -1,14 +1,31 @@
+    ```
+- [ ] Push the new derivatives to the remote storage.
+    ```shell
+    datalad push --to ria-storage
+    datalad push --to origin
````
```suggestion
    datalad push --to ria-storage
    datalad push --to origin
```

I would suggest we configure `origin` to *depend on* `ria-storage`. That way there's less room for mistakes where we forget to push to `ria-storage`. WDYT?

oesteban commented on docs/data-management/post-session.md:
```diff
@@ -212,3 +212,7 @@ As new sessions are collected, the corresponding BIDS structures MUST be saved w
     !!! danger "Always double-check that data in the annex are uploaded to the RIA store"
+
+### Run MRIQC on the data and assess their quality
+Proceed with the quality assessment of the data by following the steps detailed in the [next section](./mriqc.md).
+Checking the quality of the data short after they are acquired increases the likelihood of catching systematic artifacts and reduces the risk of the latter spreading throughout the whole dataset. It also spreads the burden of visual inspection over time, avoiding to overwhelm the rater with lots of images to rate in a short time span, which in turn reduces decision fatigue.
```
```suggestion
Checking the data quality shortly after they are acquired increases the likelihood of catching systematic artifacts early enough to avert spreading throughout the whole dataset.
It also modulates the burden of visual inspection over time, such that we avoid overwhelming raters with outbursts of images to assess.
Better pacing in rating throughput also contributes to reducing raters' attrition and fatigue.

- [ ] Screen all the unprocessed data and assess them as described in the [next section](./mriqc.md).
```

oesteban commented on docs/data-management/post-session.md:
```diff
@@ -212,3 +212,7 @@ As new sessions are collected, the corresponding BIDS structures MUST be saved w
 
     !!! danger "Always double-check that data in the annex are uploaded to the RIA store"
+
+### Run MRIQC on the data and assess their quality
```
```suggestion
### Visual assessment of unprocessed data with *MRIQC*
```

oesteban commented on docs/data-management/preliminary.md:
````diff
@@ -65,9 +65,104 @@ When employing high-performance computing (HPC), we provide [some specific guide
             --publish-depends ria-storage
     ```
+    
+- [ ] Create a sub-dataset to host the *MRIQC* derivatives. If you will not run *MRIQC* version 23.1.0, replace the version with the one you intend to run in the command below.
````
```suggestion
- [ ] Create a sub-dataset to host the *MRIQC* derivatives.
    Remember to set the correct version of the container (in our case {{ settings.versions.mriqc }}).
```

oesteban replied:
Remember to add the config for `{{ settings.versions.mriqc }}` in `/study-settings.yml` so that the replacement happens consistently throughout the SOPs.

Add something like:

```yaml
versions:
  mriqc: '23.1.0'
  fmriprep: 'xxxx'
  dmriprep: 'xxxx'
```

oesteban commented on docs/data-management/preliminary.md:
````diff
@@ -65,9 +65,104 @@ When employing high-performance computing (HPC), we provide [some specific guide
+- [ ] Create a sub-dataset to host the *MRIQC* derivatives. If you will not run *MRIQC* version 23.1.0, replace the version with the one you intend to run in the command below.
+    ``` shell
+    cd /data/datasets/hcph-dataset
+    datalad create -d . derivatives/mriqc-23.1.0
````
```suggestion
    datalad create -d . derivatives/mriqc-{{ settings.versions.mriqc }}
```

oesteban commented on docs/data-management/preliminary.md:
````diff
@@ -65,9 +65,104 @@ When employing high-performance computing (HPC), we provide [some specific guide
+
+                ``` bash
+                python -m pip --user -U pip
+                ```
+
+            - [ ] Install *DataLad*:
+
+                ``` bash
+                python -m pip install --user "datalad>=0.19" datalad-container
+                ```
+
+    - [ ] Check datalad is properly installed, for instance:
+
+        ``` shell
+        $ datalad --version
+        datalad 0.19.2
+        ```
+
+        ??? bug "*DataLad* crashes (*Conda* installations)"
+
+            *DataLad* may fail with the following error:
+            ``` py
+            ImportError: cannot import name 'getargspec' from 'inspect' (/home/users/cprovins/miniconda3/lib/python3.11/inspect.py)
+            ```
+
+            In such a scenario, create a *Conda* environment with a lower version of Python, and re-install datalad
+            ``` shell
+            conda create -n "datalad" python=3.10
+            conda activate datalad
+            conda install -c conda-forge datalad datalad-container
+            ```
+
+    - [ ] Configure your Git identity settings.
+
+        ``` shell
+        cd ~
+        git config --global --add user.name "Jane Doe"
+        git config --global --add user.email doe@example.com
+        ```
+
````
I'd prefer to keep this at the bottom where it was originally. Also, this change is unrelated to the PR.

```suggestion
```

oesteban replied:
Okay, I see the point now - there's a similar admonition for workstation installation. Let's address this somewhere else.

oesteban replied:
Actually, I don't see the point -- but still think we should defer this to a separate conversation.

oesteban commented on docs/data-management/preliminary.md:
````diff
@@ -133,93 +228,4 @@ When a new session is added, your *DataLad* dataset will remain at the same poin
-
-            ``` bash
-            python -m pip --user -U pip
-            ```
-
-        - [ ] Install *DataLad*:
-
-            ``` bash
-            python -m pip install --user "datalad>=0.19" datalad-container
-            ```
-
-- [ ] Check datalad is properly installed, for instance:
-
-    ``` shell
-    $ datalad --version
-    datalad 0.19.2
-    ```
-
-    ??? bug "*DataLad* crashes (*Conda* installations)"
-
-        *DataLad* may fail with the following error:
-        ``` py
-        ImportError: cannot import name 'getargspec' from 'inspect' (/home/users/cprovins/miniconda3/lib/python3.11/inspect.py)
-        ```
-
-        In such a scenario, create a *Conda* environment with a lower version of Python, and re-install datalad
-        ``` shell
-        conda create -n "datalad" python=3.10
-        conda activate datalad
-        conda install -c conda-forge datalad datalad-container
-        ```
-
-- [ ] Configure your Git identity settings.
-
-    ``` shell
-    cd ~
-    git config --global --add user.name "Jane Doe"
-    git config --global --add user.email doe@example.com
-    ```
-
````
```suggestion
### HPC users - instructions to install *DataLad*
When HPC is planned for processing, *DataLad* will be required on that system(s).

- [ ] Start an interactive session on the HPC cluster

    ??? warning "Do not run the installation of *Conda* and *DataLad* in the login node"
    
        HPC systems typically recommend using their login nodes only for tasks related to job submission, data management, and preparing jobscripts.
        Therefore, the execution of resource-intensive tasks such as *fMRIPrep* or building containers on login nodes can negatively impact the overall performance and responsiveness of the system for all users.
        Interactive sessions are a great alternative when available and **should** be used when creating the *DataLad* dataset.
        For example, in the case of systems operating SLURM, the following command would open a new interactive session:
        ```
        srun --nodes=1 --ntasks-per-node=1 --time=01:00:00 --pty bash -i
        ```

- [ ] Install *DataLad*.
    Generally, the most convenient and user-sandboxed installation (i.e., without requiring elevated permissions) can be achieved by using *Conda*, but other alternatives (such as *lmod*) can be equally valid:

    === "Install *DataLad* with *Conda*"
        - [ ] Get and install *Conda* if it is not already deployed in the system:
            ``` shell
            wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
            bash Miniconda3-latest-Linux-x86_64.sh
            ```
        - [ ] Install *DataLad*:
            ``` shell
            conda install -c conda-forge -y "datalad>=0.19" datalad-container
            ```

    === "Install *DataLad* in HPC with *lmod* enabled"
        - [ ] Check the availability and dependencies for a specific Python version (here we check 3.8.2):
            ``` bash
            module spider Python/3.8.2
            ```
        - [ ] Load Python (please note `ml` below is a shorthand for `module load`)
            ``` bash
            ml GCCcore/9.3.0 Python/3.8.2
            ```
        - [ ] Update *pip*:
            ``` bash
            python -m pip --user -U pip
            ```
        - [ ] Install *DataLad*:
            ``` bash
            python -m pip install --user "datalad>=0.19" datalad-container
            ```

- [ ] Check datalad is properly installed, for instance:
    ``` shell
    $ datalad --version
    datalad 0.19.2
    ```

    ??? bug "*DataLad* crashes (*Conda* installations)"

        *DataLad* may fail with the following error:
        ``` py
        ImportError: cannot import name 'getargspec' from 'inspect' (/home/users/cprovins/miniconda3/lib/python3.11/inspect.py)
        ```
        In such a scenario, create a *Conda* environment with a lower version of Python, and re-install datalad
        ``` shell
        conda create -n "datalad" python=3.10
        conda activate datalad
        conda install -c conda-forge datalad datalad-container
        ```

- [ ] Configure your Git identity settings.
    ``` shell
    cd ~
    git config --global --add user.name "Jane Doe"
    git config --global --add user.email doe@example.com
    ```

```

oesteban replied:
(I'd prefer you undo the changes on your branch rather than accepting this comment btw)

oesteban committed:
Apply suggestions from code review

oesteban committed:
fix: address several issues in PR

- [x] Roll back unrelated changes (datalad installation on clusters)
- [x] Move container registration to preliminary in data management.
    - [x] Add Docker alternative
- [x] Minor edits and clarifications.

oesteban reviewed (approved):
I'm green-ticking this but we need to come back to two aspects:

- [ ]  The actual datalad containers registration command lines (we need to check them, but I'm skeptical they will work as they stand).
- [ ] I think we want to have a separate datalad dataset for the derivatives, and only when we are ready to release derivatives, add it to the main dataset as a subdataset, but adding it in the beginning will only cause problems.
